### ISSUE TITLE
Transforming array elements into strings (same behavior for object properties that are arrays) ...

### ISSUE THREAD

**Issue by ClaudiuOctavianMuresan** [SAP/cf-nodejs-logging-support#21]
When passing an object that contains fields as arrays or even an array as a custom field we get a mutation from array type to string type (in case of object fields) or array elements are converted to strings This happens from versions > 1.1.2

**Code used for testing**
```
const logger= require('cf-nodejs-logging-support');
const a = [ { x: [1,2], y: 2, z: false, t: {}}];
logger.logMessage('info', 'Test', a);
console.log(typeof a[0], typeof a[0].x, typeof a[0].y, typeof a[0].z, typeof a[0].t);
```

**Version 1.1.2 output**
`{"written_at":"2018-05-09T10:02:09.876Z","written_ts":446547752869439,"component_type":"application","component_id":"-","component_name":"-","component_instance":"0","layer":"[NODEJS]","space_name":"-","space_id":"-","source_instance":"0","container_id":"-","logger":"nodejs-logger","msg":"Test:","type":"log","custom_fields":[{"x":[1,2],"y":2,"z":false,"t":{}}],"level":"info"}`
**object object number boolean object**

**Version 3.0.4 output**
`{"component_type":"application","component_id":"-","component_name":"-","component_instance":"0","layer":"[NODEJS]","organisation_id":"-","organisation_name":"-","space_name":"-","space_id":"-","container_id":"-","logger":"nodejs-logger","source_instance":"0","written_at":"2018-05-09T10:11:19.763Z","written_ts":447097639918326,"msg":"Test:","type":"log","custom_fields":["{\"x\":[1,2],\"y\":2,\"z\":false,\"t\":{}}"],"level":"info"}`
**string undefined undefined undefined undefined**

**Comment by christiand93** [comment]
The custom_fields field is meant to be a flat object of key-value pairs and not an array. We decided to write all values as string in order to avoid problems with varying types.

**Comment by ClaudiuOctavianMuresan** [comment]
Makes sense. Maybe the example is not clear enough. If I have an object having a field of array type and I log it and want to use it again, the field is no longer an array - it is a string. Isn't there a way to not have this behavior, maybe deep cloning or some sort?

**Comment by christiand93** [comment]
Am I correct, if I asssume you have no Problem with writing your Object as string in your log and want to preserve the object you are logging without any side effects from our library?

We will fix this issue asap.

**Comment by ClaudiuOctavianMuresan** [comment]
Thanks. You are correct.
For now we'll stick with 1.1.2 version. 
We do want to upgrade because the performance is better with newer versions like 3.0.4.

**Comment by christiand93** [comment]
We fixed this issue in version 3.0.5

**Comment by ClaudiuOctavianMuresan** [comment]
👍
